src/matcha_ml/services/azure_service.py

```diff
@@ -251,6 +251,19 @@ def resource_group_state(
 
         return None
 
+    def resource_group_exists(self, resource_group_name: str) -> bool:
+        """Checks if an Azure resource group exists.
+
+        Args:
+            resource_group_name (str): Name of the Azure resource group to check
+
+        Returns:
+            bool: True, if the resource group exists
+        """
+        rg_state = self.resource_group_state(resource_group_name)
+
+        return type(rg_state) is ProvisionState
```

Comment:
Nit: `isinstance` would be preferred over using `type`.  As `isinstance` is more generic.